trio/_tests/check_type_completeness.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python3
+# this file is not run as part of the tests, instead it's run standalone from check.sh
+import subprocess
+import json
+from pathlib import Path
+from typing import Any
+import sys
+import argparse
+
+RES_FILE = Path(__file__).parent / "verify_types.json"
+failed = False
+
+
+def run_pyright():
+    return subprocess.run(
+        ["pyright", "--verifytypes=trio", "--outputjson", "--ignoreexternal"],
+        capture_output=True,
+    )
+
+
+def main(args: argparse.Namespace) -> int:
+    print("*" * 20, "\nChecking type completeness hasn't gone down...")
+    current_dict: dict[str, Any] = {}
+    last_dict: dict[str, Any] = {}
+
+    def check_less_than(key, /, invert=False):
+        global failed
+        current = current_dict[key]
+        last = last_dict[key]
+        assert isinstance(current, (float, int))
+        assert isinstance(last, (float, int))
+        if current == last:
+            return
+        if (current > last) ^ invert:
+            failed = True
+            print("ERROR: ", end="")
+        if isinstance(current, float):
+            strcurrent = f"{current:.4}"
+            strlast = f"{last:.4}"
+        else:
+            strcurrent = str(current)
+            strlast = str(last)
+        print(
+            f"{key} has gone {'down' if current<last else 'up'} from {strlast} to {strcurrent}"
+        )
+
+    def check_zero(key):
+        if current_dict[key] != 0:
+            global failed
```

Comment:
My personal style involves putting these right under the function definition.

Reply:
even when they're accessing shared-state variables? Can def agree with putting the `print` below it, but makes sense to me to define the variables they'll use before defining the functions.

Reply:
Moved them out and got rid of shared-state variables

Reply:
Ah sorry, I meant putting it above the if statement for this comment in particular! All's good though.